feat(registry): add latexindent formatter

jackwoodside opened the pull request:
Adding [latexindent](https://github.com/cmhughes/latexindent.pl) formatter.

Closes #261

williamboman commented on lua/mason-registry/latexindent/init.lua:
```diff
@@ -0,0 +1,34 @@
+                asset_file = coalesce(
+                    when(platform.is.mac_x64, "latexindent-macos"),
+                    when(platform.is.mac_arm64, "latexindent-macos"),
+                    when(platform.is.linux_arm64, "latexindent-linux"),
+                    when(platform.is.linux_x64, "latexindent-linux"),
```
```suggestion
                    when(platform.is.linux_x64_gnu, "latexindent-linux"),
```

williamboman commented on lua/mason-registry/latexindent/init.lua:
```diff
@@ -0,0 +1,33 @@
+
+return Pkg.new {
+    name = "latexindent",
+    desc = [[Perl script to add indentation to LaTeX files.]],
```
```suggestion
    desc = _.dedent [[
        latexindent.pl is a perl script to beautify/tidy/format/indent (add horizontal leading space to) code within
        environments, commands, after headings and within special code blocks.
    ]],
```

williamboman reviewed (approved):
Cool! I removed `linux_arm64` because it pointed to a x64 binary. Also added `gnu` requirement to `linux_x64` because the binary is linked to glibc

jackwoodside commented:
> Cool! I removed `linux_arm64` because it pointed to a x64 binary. Also added `gnu` requirement to `linux_x64` because the binary is linked to glibc

Thanks, I'll keep that in mind for any future additions
